components/sidebar_right/sidebar_right.jsx

```diff
@@ -73,12 +73,25 @@ export default class SidebarRight extends React.PureComponent {
     }
 
     handleShortcut = (e) => {
-        if (Utils.cmdOrCtrlPressed(e) && Utils.isKeyPressed(e, Constants.KeyCodes.PERIOD)) {
-            e.preventDefault();
-            if (this.props.isOpen) {
-                this.props.actions.closeRightHandSide();
-            } else {
-                this.props.actions.openAtPrevious(this.previous);
+        if (Utils.cmdOrCtrlPressed(e)) {
+            if (e.shiftKey && Utils.isKeyPressed(e, Constants.KeyCodes.PERIOD)) {
+                e.preventDefault();
+                if (this.props.isOpen) {
+                    if (this.props.isExpanded) {
+                        this.props.actions.setRhsExpanded(false);
+                    } else {
+                        this.props.actions.setRhsExpanded(true);
+                    }
+                } else {
+                    this.props.actions.openAtPrevious(this.previous);
+                }
+            } else if (Utils.isKeyPressed(e, Constants.KeyCodes.PERIOD)) {
+                e.preventDefault();
+                if (this.props.isOpen) {
+                    this.props.actions.closeRightHandSide();
+                } else {
+                    this.props.actions.openAtPrevious(this.previous);
+                }
```

Comment:
My only small suggestion to improve the code readability is to :- 
```
if (Utils.cmdOrCtrlPressed(e) && e.shiftKey && Utils.isKeyPressed(e, Constants.KeyCodes.PERIOD))) {
}

Reply:
I could see going either way with it. Nesting is nice to avoid repeating ourselves and because it makes `else` cases clearer (like its not immediately obvious to me that its when shift isn't held in the second case of your example above), but your changes do make it easier to see exactly which keys are required for each case